client.py

```diff
@@ -1,6 +1,7 @@
 from core import *
 from threading import Thread
-
+import os.path as osp
```

Comment:
I see you're using the os.path module, I was checking this [article](https://medium.com/@ageitgey/python-3-quick-tip-the-easy-way-to-deal-with-file-paths-on-windows-mac-and-linux-11a072b58d5f) where they provide a better solution  (using the pathlib library) if you want to implement that instead.